pkg/operator/resources/batchapi/logging.go

```diff
@@ -43,17 +43,31 @@ func operatorLogStream(jobKey spec.JobKey) string {
 
 // Checks if log group exists before creating it
 func ensureLogGroupForAPI(apiName string) error {
-	output, err := config.AWS.CloudWatchLogs().DescribeLogGroups(&cloudwatchlogs.DescribeLogGroupsInput{
-		Limit:              aws.Int64(1),
-		LogGroupNamePrefix: aws.String(logGroupNameForAPI(apiName)),
-	})
+	apiNameLogGroup := logGroupNameForAPI(apiName)
+	logGroupExists := false
+
+	err := config.AWS.CloudWatchLogs().DescribeLogGroupsPages(
```

Comment:
I know this is the implementation we had discussed during the call a while back. I think we can do an exact match. There is a function called `DoesLogGroupExist` in our `lib/aws` package which can be used to determine if a log group exists or not. That function can be used instead of doing a prefix search.

My apologies!

Reply:
No worries! :) With `DoesLogGroupExist` in, the `ensureLogGroupForAPI` function is now simpler.